src/libsystemd/sd-device/sd-device.c

```diff
@@ -2542,14 +2561,20 @@ _public_ int sd_device_get_sysattr_value(sd_device *device, const char *sysattr,
         assert(k > 0);
 
         if (ret_value && r >= 0)
-                *ret_value = value;
+                *ret_value = ret_size ? value : value_stripped;
```

Comment:
Then, when `ret_size` is `NULL`, please get the truncated string from the cache.